gp/primitives.py

```diff
@@ -34,7 +37,10 @@ def __init__(self, primitives):
 
         self._determine_max_arity()
 
-    def _determine_max_arity(self):
+    def __iter__(self) -> Iterator:
+        return iter([self[i] for i in range(len(self._primitives))])
```

Comment:
this is a potentially dangerous statement: it's important that the order of primitives is exactly conserved from how this object was initialized (ll29) otherwise the mapping from function id to function changes when an individual is cloned (where this function is used). maybe we should use an `OrderedDict` and just return `self._primitives.values()` here? at minimum this line should have a "WARNING" comment that reminds us of the importance of the correct order

Reply:
`self._primitives` is created with this code:
```
for i in range(len(primitives)):
            self._primitives[i] = primitives[i]
```
Thus the elements are with increasing index `0, 1, 2,...` into the dict, which exactly matches the way they are retrieved here (with a `for i in range(len(...)`. So I think this is safe here.

Reply:
It could be written in a better way, but that is out of scope of this PR. Please create an issue if you think it's necessary.

Reply:
it's not that your operation is unsafe, but someone who wants to do "smart" refactoring might change this for example to iterating over the entries of `_primitives`. created #107 to address this